packages/runtime/container-runtime/src/runWhileConnectedCoordinator.ts

```diff
@@ -68,9 +68,9 @@ export class RunWhileConnectedCoordinator implements ICancellableSummarizerContr
         if (this.runtime.connected) {
             this.everConnected = true;
         } else if (this.runtime.disposed) {
+            this.everConnected = true;
```

Comment:
Not saying that this is wrong but I'm a bit confused about why this is a safe inference? Would this whole thing be moot if the container runtime was never connected?

Reply:
I like to have fewer different states to reason. Given that disposed and previously connected, and disposed and never connected should be treated the same by this code, I prefer to collapse these two states in terms of how they are represented in states of this class.

That said, I'm realizing that it's not actually true and current logic still has pretty big gap - it never listens on "disposed" state. :) So changes are required.

Reply:
And opened https://github.com/microsoft/FluidFramework/issues/7338 to follow up with proper UTs here, as obviously it was a huge gap in this logic.